### PR TITLE
Add new rule: "SHOULD select appropriate one of date or date-time format"

### PR DESCRIPTION
**Pull request by vadimturkov**
Closing https://github.com/zalando/restful-api-guidelines/issues/797.

### CODE REVIEW

File: chapters/data-formats.adoc [hunk 1]
```diff
@@ -124,6 +124,21 @@ issues with precision, e.g. whether to represent a timestamp as 1460062925,
 more effort to parse, avoid this ambiguity.
 
 
+[#255]
+== {SHOULD} select appropriate one of date or date-time format
+
+When choosing between `date` and `datetime` formats you should take into account the following:
+
+* `date` should be used for properties that are not time-sensitive or where precise timing is not required,
```
Comment (tfrauenstein) [inline chapters/data-formats.adoc:132]:
```suggestion
* `date` should be used for properties where no exact point in time is not required and day time-range is sufficient,
```

File: chapters/data-formats.adoc [hunk 2]
```diff
@@ -124,6 +124,21 @@ issues with precision, e.g. whether to represent a timestamp as 1460062925,
...
+for instance, document dates, birthdays, ETAs (estimated time of arrival).
+Normally date without time implies the time period from midnight to midnight in the local time zone.
```
Comment (tfrauenstein) [inline chapters/data-formats.adoc:134]:
```suggestion
Without further context, `date` without time implies the time period from midnight to midnight in the local time zone.
```

File: chapters/data-formats.adoc [hunk 3]
```diff
@@ -124,6 +124,21 @@ issues with precision, e.g. whether to represent a timestamp as 1460062925,
...
+Since time zone information is absent in the property value, it must be documented and can be also provided
```
Comment (tfrauenstein) [inline chapters/data-formats.adoc:135]:
```suggestion
However, the timezone information can be also provided
```

File: chapters/data-formats.adoc [hunk 4]
```diff
@@ -124,6 +124,21 @@ issues with precision, e.g. whether to represent a timestamp as 1460062925,
...
+as an additional context via other fields indicating location.
```
Comment (tfrauenstein) [inline chapters/data-formats.adoc:136]:
```suggestion
as an additional context information via other fields indicating location.
```

File: chapters/data-formats.adoc [hunk 5]
```diff
@@ -124,6 +124,21 @@ issues with precision, e.g. whether to represent a timestamp as 1460062925,
...
+* `datetime` should be used in all other cases to indicate instants/moments in time.
```
Comment (tfrauenstein) [inline chapters/data-formats.adoc:137]:
```suggestion
* `datetime` should be used in all other cases where an exact point in time is required, 
for instance, datetimes for supplier advice, specific processing events, fast delivery planning dates.
```

File: chapters/data-formats.adoc [hunk 6]
```diff
@@ -124,6 +124,21 @@ issues with precision, e.g. whether to represent a timestamp as 1460062925,
...
+This format requires the explicit time zone offset to be provided,
```
Comment (tfrauenstein) [inline chapters/data-formats.adoc:138]:
```suggestion
As required in <<169>>,  `datetime` requires the explicit time zone offset to be provided,
```

**Review by tfrauenstein — COMMENTED** [review]
Thx a lot for the proposal. LGTM -- provided editorial change proposals.

**Comment by vadimturkov** [comment]
@tfrauenstein thanks for the suggestions. Applied.

**Comment by tfrauenstein** [comment]
👍

**Comment by vadimturkov** [comment]
@ePaul you marked this PR with `zally-change` label, does it mean we need to adjust Zally?

**Comment by ePaul** [comment]
@vadimturkov Hmm, that would be the meaning of this label, but looking at the text again, I don't remember anymore why I did this.

**Comment by ePaul** [comment]
:+1:

**Pull request merged**